homeassistant/components/home_connect/coordinator.py

```diff
@@ -73,6 +75,29 @@ def __init__(
             str, dict[EventKey, list[Callable[[Event], Coroutine[Any, Any, None]]]]
         ] = {}
 
+    @cached_property
+    def context_listeners(self) -> dict[tuple[str, EventKey], list[CALLBACK_TYPE]]:
+        """Return a dict of all listeners registered for a given context."""
+        listeners: dict[tuple[str, EventKey], list[CALLBACK_TYPE]] = {}
+        for listener, context in list(self._listeners.values()):
+            assert isinstance(context, tuple)
+            listeners.setdefault(context, []).append(listener)
+        return listeners
+
+    @callback
+    def async_add_listener(
+        self, update_callback: CALLBACK_TYPE, context: Any = None
+    ) -> Callable[[], None]:
+        """Listen for data updates."""
+        remove_listener = super().async_add_listener(update_callback, context)
```

Comment:
I think we need to invalidate the cached property also when adding a new listener, not only when removing a listener. See the nibe heatpump example.